src/DurableTask.Core/History/TaskScheduledEvent.cs

```diff
@@ -84,5 +85,11 @@ public TaskScheduledEvent(int eventId)
         /// </summary>
         [DataMember]
         public DistributedTraceContext? ParentTraceContext { get; set; }
+        
+        /// <summary>
+        /// Gets or sets a dictionary of tags of string, string
+        /// </summary>
+        [DataMember]
+        public IDictionary<string, string> Tags { get; set; } = new Dictionary<string, string>();
```

Comment:
Make this nullable by default with no dictionary, so that we're not allocating this additional memory for each of these objects when we don't need it. Then it will also match the `Tags` property for `ExecutionStartedEvent`.
```suggestion
        public IDictionary<string, string>? Tags { get; set; }
```